File: Backend/src/main/java/com/ExpenseOS/Backend/service/UserService.java
package com.ExpenseOS.Backend.service;

import com.ExpenseOS.Backend.dto.user.UserResponse;
import com.ExpenseOS.Backend.entity.User;
import com.ExpenseOS.Backend.exception.ResourceNotFoundException;
import com.ExpenseOS.Backend.repository.UserRepository;
import lombok.RequiredArgsConstructor;
import org.springframework.stereotype.Service;

@Service
@RequiredArgsConstructor
public class UserService {

    private final UserRepository userRepository;

    public UserResponse getCurrentUser(String email){

        User user = userRepository.findByEmail(email)
                .orElseThrow(() ->
                        new ResourceNotFoundException("User not found"));

        return UserResponse.builder()
                .id(user.getId())
                .name(user.getName())
                .email(user.getEmail())
                .build();
    }
}

File: Mobile/src/services/userService.ts
import { apiClient } from './apiClient';

export interface UserResponse {
  id: number;
  name: string;
  email: string;
}

export const userService = {
  getCurrentUser: async (): Promise<UserResponse> => {
    const response = await apiClient.get<UserResponse>('/users/me');
    return response.data;
  },
};
